pgadmin4-17.yaml

```diff
@@ -0,0 +1,164 @@
+    match: '\.'
+    replace: "_"
+    to: mangled-package-version
+
+pipeline:
+  - uses: git-checkout
+    with:
+      expected-commit: 9e58717321a434519875773707df78c020bcde40
+      repository: https://github.com/pgadmin-org/pgadmin4
+      tag: REL-${{vars.mangled-package-version}}
+
+  - name: Building Web Assests
+    runs: |
+      cd web
+      export CPPFLAGS="$CPPFLAGS -DPNG_ARM_NEON_OPT=0"
+      npm install -g corepack
+      corepack enable
+      yarn set version berry
+      yarn set version 3
+      yarn install
+      yarn run bundle
+
+  - name: Build package
+    runs: |
+      # Setup Python virtual environment
+      python3 -m venv venv
+      source venv/bin/activate
+
+      # Upgrade basic tools
+      pip install --upgrade pip
+
+      # Install Python dependencies
+      pip install -r requirements.txt
+      pip install sphinx sphinxcontrib-youtube
+      pip install -r web/regression/requirements.txt
+
+      # Build documentation and ignore warnings
+      make -C docs/en_US -f Makefile.sphinx SPHINXOPTS='' html || echo "Documentation warnings ignored"
+
+      # Build Python package
+      make pip
+
+      mkdir -p ${{targets.destdir}}/pgadmin4
+      mkdir -p ${{targets.destdir}}/var/lib/pgadmin
+      mkdir -p ${{targets.destdir}}/var/log/pgadmin
+
+      # Copy in the code and docs
+      cp -R web/ ${{targets.destdir}}/pgadmin4/
+      cp -R docs/ ${{targets.destdir}}/pgadmin4/
+      cp -R venv ${{targets.destdir}}/pgadmin4/
+      cp ./pkg/docker/run_pgadmin.py ${{targets.destdir}}/pgadmin4/
+      cp ./pkg/docker/gunicorn_config.py ${{targets.destdir}}/pgadmin4/
+      cp ./pkg/docker/entrypoint.sh ${{targets.destdir}}/entrypoint.sh
+
+      # License files
+      cp -R ./LICENSE ${{targets.destdir}}/pgadmin4/
+      cp -R ./DEPENDENCIES ${{targets.destdir}}/pgadmin4/
+
+      # sed command causes errors trying to edit .pyc files in pycache
+      rm -rf ${{targets.destdir}}/pgadmin4/venv/bin/__pycache*
+      sed -i "s|/home/build|/pgadmin4|g" ${{targets.destdir}}/pgadmin4/venv/bin/*
+      # allow site-packages
+      sed -i "s|include-system-site-packages = false|include-system-site-packages = true|g" ${{targets.destdir}}/pgadmin4/venv/pyvenv.cfg
+
+test:
+  environment:
+    contents:
+      packages:
+        - postgresql
+        - postgresql-client
+        - python3
+        - py3-pip
+        - shadow
+        - sudo-rs
+        - curl
+    environment:
+      PGADMIN_SETUP_EMAIL: admin@gmail.com
+      PGADMIN_SETUP_PASSWORD: admin123
+      PGDATA: /tmp/test_db
+      PGUSER: wolfi
+  pipeline:
+    - name: "Test database creation"
+      runs: |
+        useradd $PGUSER
+        sudo -u $PGUSER initdb -D /tmp/test_db
+        sudo -u $PGUSER pg_ctl -D /tmp/test_db -l /tmp/logfile start
+        createdb testdb
+        psql -lqt | cut -d \| -f 1 | grep -qw testdb
+    - name: "Run pgAdmin4 Initialization"
+      runs: |
+        cd /pgadmin4
+        source venv/bin/activate
```

Comment:
We should expect our package to work without sourcing the virtual environment. We should set PYTHONPATH or run the binaries (really scripts) in the bin folder pointing to the vendored Python install